### PR TITLE
feat(dot/rpc) Add `system_syncState` rpc call

### PR DESCRIPTION
**Pull request by EclesioMeloJunior**
## Changes

- Add `system_syncState` rpc call
- Add `HighestBlock` and `StartingBlock` to the network API interface

## Tests

```
go test -timeout 30s -run ^TestSyncState$ github.com/ChainSafe/gossamer/dot/rpc/modules
```

## Issues

- Related to #731 

## Screenshots

1. Was made a `system_syncState` rpc call to a Gossamer node running with flags `--roles 1 --chain Polkadot`

image

### CODE REVIEW

File: dot/network/service.go [hunk 1]
```diff
@@ -674,3 +674,13 @@ func (s *Service) Peers() []common.PeerInfo {
 func (s *Service) NodeRoles() byte {
 	return s.cfg.Roles
 }
+
+// HighestBlock returns the highest known block number
+func (s *Service) HighestBlock() int64 {
+	return s.syncQueue.goal
+}
+
+// StartingBlock return the starting block number
```
Comment (noot) [inline dot/network/service.go:683]: starting block number that's currently being synced

**Review by noot — APPROVED** [review]
nice :D

File: dot/network/service.go (hunk 1 above)
Comment (EclesioMeloJunior) [reply dot/network/service.go]: updated!

File: dot/network/service.go [hunk 2]
```diff
@@ -674,3 +674,13 @@ func (s *Service) Peers() []common.PeerInfo {
 func (s *Service) NodeRoles() byte {
 	return s.cfg.Roles
 }
+
+// HighestBlock returns the highest known block number
+func (s *Service) HighestBlock() int64 {
```
Comment (timwu20) [inline dot/network/service.go:679]: any reason why this isn't a `uint64`?

Comment (EclesioMeloJunior) [reply dot/network/service.go:679]: I just keep the return type as the type of `syncQueue.currStart`,

**Review by timwu20 — APPROVED** [review]

**Pull request merged**